tests/testthat/test-smc_compute_rho_consensus.R

```diff
@@ -53,7 +53,7 @@ test2 <- compute_rho_consensus(
 )
 
 test_that("Output of compute_rho_consensus (CP) is OK", {
-  expect_is(test1, "tbl_df")
+  expect_is(test1, "data.frame")
```

Comment:
I'm in favor of replacing tibbles with data frames, but I am a bit worried about this breaking backwards compatibility. It looks like we're not following SemVer on this package, but maybe this kind of change is large enough to trigger a major version increase (so BayesMallows 2.0.0 would be the first release including this PR)?

Reply:
Good point, although I'm not entirely sure what I think. 🤔 

A comparison between tibbles and dataframes is [here](https://www.rstudio.com/blog/tibble-1-0-0/). What I can foresee as a possible problem is if someone uses the result from `compute_rho_consensus` and relies on the particular properties that distinguish tibbles from dataframes. Essentially, this differences is the following:

``` r
df_base <- data.frame(a = 1:3)
df_tibble <- dplyr::tibble(a = 1:3)

# returns vector
df_base[, "a"]
#> [1] 1 2 3
# returns data.frame
df_base[, "a", drop = FALSE]
#>   a
#> 1 1
#> 2 2
#> 3 3
# returns tibble always
df_tibble[, "a"]
#> # A tibble: 3 × 1
#>       a
#>   <int>
#> 1     1
#> 2     2
#> 3     3

# returns NULL
df_base$b
#> NULL
# gives WARNING before returning NULL
df_tibble$b
#> Warning: Unknown or uninitialised column: `b`.
#> NULL
```

<sup>Created on 2022-05-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

As the package does not have any real reverse dependencies, this will not cause trouble for any exisiting packages. It could however cause trouble if someone has code that depends on this behavior. I'm leaning towards just changing this, put we could perhaps also give a deprecation notice for a while, before changing to `data.frames`?

Reply:
> give a deprecation notice for a while, before changing to `data.frames`

Sounds like a good compromise!

Reply:
In [this commit](https://github.com/ocbe-uio/BayesMallows/pull/211/commits/2234619b95b340697908f935fd240f8c0ccdc2ee) I made a message in  that is given once per session:

``` r
# New sessions
library(BayesMallows)
model_fit <- compute_mallows(preferences = beach_preferences,
                             save_aug = TRUE)
#> Generating transitive closure of preferences.
#> Generating initial ranking.
model_fit$burnin <- 1000
probs <- predict_top_k(model_fit)
#> Change notice: predict_top_k() now returns a data.frame, and not a tibble.
#> To get a tibble run tibble::as_tibble() on the returned object.
#> This message is given once per session.
probs <- predict_top_k(model_fit)
probs <- predict_top_k(model_fit)
```

<sup>Created on 2022-05-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
Nice touch with the once-per-session message, and props for the safe use of global variables (coincidentally, I was reading the [_R Inferno_ chapter on that](https://www.burns-stat.com/pages/Tutor/R_inferno.pdf#chapter.6) just yesterday 😆)!

Reply:
Found the solution here, and should maybe have added a link, so didn't think that out myself: https://stat.ethz.ch/pipermail/r-help//2014-August/421105.html

Reply:
Hadn't heard of R inferno, btw. Looks like a cool book :-)